src/EditorFeatures/CSharp/InlineRename/CSharpEditorInlineRenameService.cs

```diff
@@ -18,4 +26,119 @@ internal sealed class CSharpEditorInlineRenameService(
+    /// </summary>
+    /// <param name="inlineRenameInfo">Instance of <see cref="IInlineRenameInfo"/> pertinent to the rename session.</param>
+    /// <param name="inlineRenameLocationSet"><see cref="IInlineRenameLocationSet"/> of references discovered for the rename session.</param>
+    /// <param name="cancellationToken">Cancellation token</param>
+    /// <returns>Map where key indicates the kind of semantic information, and value is an array of relevant code snippets.</returns>
+    protected override async Task<ImmutableDictionary<string, ImmutableArray<string>>> GetRenameContextCoreAsync(IInlineRenameInfo inlineRenameInfo, IInlineRenameLocationSet inlineRenameLocationSet, CancellationToken cancellationToken)
+    {
+        using var _1 = PooledHashSet<TextSpan>.GetInstance(out var seen);
+        using var _2 = ArrayBuilder<string>.GetInstance(out var definitions);
+        using var _3 = ArrayBuilder<string>.GetInstance(out var references);
+        using var _4 = ArrayBuilder<string>.GetInstance(out var docComments);
+
+        foreach (var renameDefinition in inlineRenameInfo.DefinitionLocations)
+        {
+            // Find largest snippet of code that represents the definition
+            var containingStatementOrDeclarationSpan =
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<StatementSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false) ??
+                await renameDefinition.Document.TryGetSurroundingNodeSpanAsync<MemberDeclarationSyntax>(renameDefinition.SourceSpan, cancellationToken).ConfigureAwait(false);
+
+            var documentText = await renameDefinition.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            if (documentText is null)
+            {
+                continue;
+            }
+
+            // Find documentation comments using optional service
+            var symbolService = renameDefinition.Document.GetLanguageService<IGoToDefinitionSymbolService>();
+            if (symbolService is not null)
+            {
+                var textSpan = inlineRenameInfo.TriggerSpan;
+                var (symbol, _, _) = await symbolService.GetSymbolProjectAndBoundSpanAsync(
+                    renameDefinition.Document, textSpan.Start, cancellationToken)
+                    .ConfigureAwait(true);
+                var docComment = symbol?.GetDocumentationCommentXml(expandIncludes: true, cancellationToken: cancellationToken);
+                if (!string.IsNullOrWhiteSpace(docComment))
+                {
+                    docComments.Add(docComment!);
+                }
+            }
+
+            AddSpanOfInterest(documentText, renameDefinition.SourceSpan, containingStatementOrDeclarationSpan, definitions);
+        }
+
+        foreach (var renameLocation in inlineRenameLocationSet.Locations)
```

Comment:
there may be 10s of thousands of locations here.  is htere something that is limiting the amount of work we do at some point?